cookbooks/mediawiki/recipes/default.rb

```diff
@@ -22,6 +22,18 @@
 include_recipe "mysql"
 include_recipe "git"
 
+# Mediawiki Parsoid package is available from wikimedia.org repositories,
+# more information at https://www.mediawiki.org/wiki/Parsoid/Setup#Ubuntu_/_Debian
+apt_repository 'mediawiki' do
+  uri           'https://releases.wikimedia.org/debian'
+  distribution  'jessie-mediawiki'
+  components    ['main']
+  key           'AF380A3036A03444'
+  keyserver     'keyserver.ubuntu.com'
+end
```

Comment:
This is not how we configure APT in our environment - the repositories are configured via the `apt` cookbook so it's not acceptable to try and do it here as well.

The problem is you're trying to use one cookbook on it's own and that just doesn't work for any non-trivial cookbook in our setup.

Reply:
> The problem is you're trying to use one cookbook on it's own and that just doesn't work for any non-trivial cookbook in our setup.

I think that's a cop-out. All the cookbooks should work, trivial or not. If they have hidden dependencies on other cookbooks, then those dependencies should be made explicit.

>  so it's not acceptable to try and do it here as well.

I'd prefer to see a suggestion of how you would prefer to see the problem solved, rather than just saying something is "not acceptable". For example, if running the apt recipe is needed first, then you could suggest adding an 'include_recipe' statement.

Reply:
Well he hasn't really explained what he was doing - there is no test currently declared for this cookbook so presumably he was specifying something on the command line somehow?

Obviously the depedncency should be noted in `metadata.rb` though that isn't actually enough to make it run, and I'm not sure it's safe or necessarily correct or appropriate to just `include_recipe` it - our whole system is built around the roles and assumes that the base roles and the cookbooks they require will be run everywhere. Fixing (or changing if you don't accept that the current state is broken as such) that is a massive job.

Reply:
My test plan in the PR upthread explains that I was fixing some issues that I had run into when trying to run the wiki cookbook under test-kitchen. Thanks for your suggestions, I see now that the missing Apt repository is specified in a different cookbook that wasn't declared as a dependency in this one. I also noticed that it was necessary to have `recipe[munin]` in my test configuration; I’ll modify this PR to account for your suggestions.

Reply:
My point was that `.kitchen.yml` doesn't contain any tests for the wiki cookbook, or are you tryimg to add one?

Reply:
I’ll go ahead and add one — I was using the one described in my test plan above, but it sounds from your suggestions like it would be appropriate to add one like:

      - name: wiki
        run_list:
          - recipe[wiki]
          - role[wiki]

Reply:
I wouldn't expect to name both the recipe and the role like that - does the role not pull in the recipe anyway? You will also need to pull in the dependent recipes most likely - look at the otrs test for example.

I suspect the wiki one probably has quite complex dependencies which is likely why nobody has been in a hurry to tackle it.